### PR TITLE
[Bugfix:Forum] Grader failed to access Forum

### PR DESCRIPTION
**Pull request by JWei17**
### Please check if the PR fulfills these requirements:

* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant
* [ ] Screenshots are attached to Github PR if visual/UI changes were made

### What is the current behavior?

When a grader click any of the exist thread, or published a new thread. A "Frog robot"  warning will pop up.

here is the frog robot.
Screenshot 2024-04-09 162746

### What is the new behavior?
Grader are able to published a new thread or click on the exist thread with out the "Frog robot pop up".
### Other information?
Procedure:
login as  a grader.
Go to discussion forum.
Click any exist thread or publish a new thread you will see the error.

### CODE REVIEW

File: site/app/views/forum/ForumThreadView.php [hunk 1]
```diff
@@ -1199,7 +1182,7 @@ public function createPost(array $thread, array $post, $unviewed_posts, $first,
             }
         }
 
-        if ($this->core->getUser()->getGroup() == 4) {
```
Comment (DarthNyan) [inline site/app/views/forum/ForumThreadView.php:1202]: This is the source of your PHP lint failure: you fixed a loose comparison so PHPstan complains that not enough loose comparisons were found. To fix this, [rebuild the baseline](https://submitty.org/developer/testing/linting_static_analysis).

File: site/app/views/forum/ForumThreadView.php [hunk 2]
```diff
@@ -1138,31 +1138,14 @@ public function createPost(array $thread, array $post, $unviewed_posts, $first,
...
-            $jscriptAnonFix = json_encode($jscriptAnonFix);
-
-            $post_user_info = [
-                "info_name" => $info_name,
-                "visible_user_json" => $visible_user_json,
-                "jscriptAnonFix" => $jscriptAnonFix,
-                "pronouns" => $pronouns,
-                "display_pronouns" => $display_pronouns
-            ];
-        }
```
Comment (DarthNyan) [inline site/app/views/forum/ForumThreadView.php:1157]: For other reviewers: this code is reused in a separate if block below, making it redundant here.

**Review by DarthNyan — APPROVED** [review]
Looks good to me, though see my comment about rebuilding the baseline.

**Review by cjreed121 — COMMENTED** [review]
Code looks good. Will approve after PHP lint is fixed.

**Review by cjreed121 — APPROVED** [review]
Code looks good

**Comment by bmcutler** [comment]
@JWei17 please fix merge conflicts

**Pull request merged**